projects/T-beam/src/gps.cpp

```diff
@@ -1,26 +1,38 @@
 #include "gps.h"
+#include "TinyGPS++.h"
 
-namespace gps {
+namespace {
+TinyGPSPlus tiny_gps;
+gps::Data gps_data;
+bool data_available = false;
+} // namespace
 
+namespace gps {
 HardwareSerial GPSSerial(GPS_SERIAL_NUM);
 
 void init() { GPSSerial.begin(9600, SERIAL_8N1, GPS_RX_PIN, GPS_TX_PIN); }
 
-void gps_loop(void* pvParameters) {
-  String nmea = "";
+bool data_is_available() { return data_available; }
+Data get_gps_data() { return gps_data; }
+
+void gps_task(void *pvParameters) {
   for (;;) {
-    if (GPSSerial.available()) {
-      char c = GPSSerial.read();
-      nmea += c;
-    } else {
-      if (nmea != "") {
-        Serial.println(nmea);
-        nmea = "";
-      } else
-        Serial.println("------------");
-      vTaskDelay(pdMS_TO_TICKS(100));
+    while (GPSSerial.available()) {
+      bool sentence_finished = tiny_gps.encode(GPSSerial.read());
+
+      if (sentence_finished && tiny_gps.location.isValid() &&
+          tiny_gps.time.isValid()) {
+        gps_data.lat = tiny_gps.location.lat();
+        gps_data.lng = tiny_gps.location.lng();
+        gps_data.time.hours = tiny_gps.time.hour();
+        gps_data.time.minutes = tiny_gps.time.minute();
+        gps_data.time.seconds = tiny_gps.time.second();
+
+        data_available = true;
+      }
     }
+
+    vTaskDelay(pdMS_TO_TICKS(100));
```

Comment:
are we 100% sure this value will always be 100? Is there any benefit to making it a variable?

Reply:
At the moment it's essentially a placeholder, no reason to refactor into a variable. Possibly, we could use interrupts to read the GPS serial, but right now the focus was on getting basic functionality